include/spi.h

```diff
@@ -83,6 +83,35 @@ typedef enum spi_bit_order {
 	SPI_BIT_ORDER_LSB_FIRST = 1,
 } spi_bit_order;
 
+#define SPI_MSG(_tx_buff, _rx_buff, _bytes_number, _cs_change, _next) \
```

Comment:
Secondly, I'd like to see an example of 3-4 messages initialized one after another, to me it seems it's going to be a pain in the ass for the user to remember to initialize .next....

Reply:
I used it like: 
![image](https://user-images.githubusercontent.com/16192750/114030845-72da9f00-9883-11eb-962d-0c80b882ed94.png)

Reply:
So if I understand this correctly you'd need to declare:
msg1
msg2
msg3

Then do:
SPI_MSG(msg1 parameters, &msg2)
SPI_MSG(msg2 parameters, &msg3)
SPI_MSG(msg3 parameters, NULL)

All this verbosity could dissapear with a simple array like
transfers[] = {
 { msg1 parameters },
 {msg2 parameters },
 {msg3 parameters },
 };
 
I really don't like that the user needs to take care about next, that's internal implementation specifics that should be hidden. If you really want to do it with list, do it with a list, but construct your list from a user-friendly user interface such as an array.